fvtest/compilertriltest/SimplifierFoldAndTest.cpp

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2017, 2017
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+#include "JitTest.hpp"
+#include "jitbuilder_compiler.hpp"
+#include "il/Node.hpp"
+#include "infra/ILWalk.hpp"
+#include "ras/IlVerifier.hpp"
+#include "ras/IlVerifierHelpers.hpp"
+
+/* Verify the optimization was correctly applied.
+ * A test failure is triggered by using `EXPECT_*`, and
+ * compilation is stopped by returning a non-zero return code.
+ * `ADD_FAILURE` is equivalent to expecting something false.
+ *
+ * While anything can be done in `verify`, this test
+ * is implemented by traversing the IL, and checking if
+ * there are AND operations.
+ */
+class SimplifierFoldAndIlVerifier : public TR::IlVerifier
+   {
+   public:
+   int32_t verifyNode(TR::Node *node)
+      {
+      switch(node->getOpCodeValue())
+         {
+         case TR::land:
+         case TR::iand:
+         case TR::band:
+         case TR::sand:
```

Comment:
Would `node->getOpCode().isAnd()`  be a better choice here rather than the switch statement?  Right now, you're letting `TR::vand`'s through, which I don't think is what you want (and `isAnd()` is more succinct).